src/core/load-fragment.c

```diff
@@ -3903,6 +3903,117 @@ int config_parse_memory_limit(
         return 0;
 }
 
+int config_parse_device_memory_limit(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        _cleanup_free_ char *root = NULL;
+        _cleanup_strv_free_ char **domains = NULL, **capacities = NULL;
+        _cleanup_strv_free_ char **limit_domains = NULL, **limits = NULL;
+        CGroupContext *c = data;
+        uint64_t scale = CGROUP_LIMIT_MAX;
+        int r;
+
+        if (isempty(rvalue) && STR_IN_SET(lvalue, "DeviceMemoryLow",
+                                          "DeviceMemoryMin"))
+                scale = CGROUP_LIMIT_MIN;
+        else if (!isempty(rvalue) && !streq(rvalue, "infinity")) {
+
+                r = parse_permyriad(rvalue);
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r, "Invalid device memory limit '%s', ignoring: %m", rvalue);
+                        return 0;
+                }
+                scale = r;
+
+                if (scale > 10000U) {
+                        log_syntax(unit, LOG_WARNING, filename, line, 0, "Device memory limit '%s' out of range, ignoring.", rvalue);
+                        return 0;
+                }
+        }
+
+        r = cg_get_root_path(&root);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to determine root cgroup, ignoring device memory limit: %m");
+                return 0;
+        }
+
+        r = cg_get_all_keyed_attributes("dmem", "", "dmem.capacity", &domains, &capacities);
```

Comment:
Uh... I do not think it is better to calculate the size when parsing unit setting.
Please calculate when apply.